hapi-fhir-jpaserver-cql/src/main/java/ca/uhn/fhir/cql/common/evaluation/LibraryLoader.java

```diff
@@ -0,0 +1,105 @@
+package ca.uhn.fhir.cql.common.evaluation;
+
+import org.cqframework.cql.cql2elm.CqlTranslator;
+import org.cqframework.cql.cql2elm.CqlTranslatorException;
+import org.cqframework.cql.cql2elm.CqlTranslatorOptions;
+import org.cqframework.cql.cql2elm.LibraryManager;
+import org.cqframework.cql.cql2elm.ModelManager;
+import org.cqframework.cql.elm.execution.Library;
+import org.cqframework.cql.elm.execution.VersionedIdentifier;
+
+import javax.xml.bind.JAXBException;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static ca.uhn.fhir.cql.common.helper.TranslatorHelper.errorsToString;
+import static ca.uhn.fhir.cql.common.helper.TranslatorHelper.getTranslator;
+import static ca.uhn.fhir.cql.common.helper.TranslatorHelper.readLibrary;
+
+public class LibraryLoader implements org.opencds.cqf.cql.engine.execution.LibraryLoader {
+	private LibraryManager libraryManager;
+	private ModelManager modelManager;
+	private Map<String, Library> libraries = new HashMap<>();
+
+	// private static final Logger logger =
+	// LoggerFactory.getLogger(LibraryLoader.class);
+
+	public Collection<Library> getLibraries() {
+		return this.libraries.values();
+	}
+
+	public LibraryManager getLibraryManager() {
+		return this.libraryManager;
+	}
+
+	public ModelManager getModelManager() {
+		return this.modelManager;
+	}
+
+	public LibraryLoader(LibraryManager libraryManager, ModelManager modelManager) {
+		this.libraryManager = libraryManager;
+		this.modelManager = modelManager;
+	}
+
+	private Library resolveLibrary(VersionedIdentifier libraryIdentifier) {
+		if (libraryIdentifier == null) {
+			throw new IllegalArgumentException("Library identifier is null.");
+		}
```

Comment:
This is just a nitpick - But it's cleaner to read I find and has the advantage of not dropping the code coverage numbers which I always like.

```suggestion
		Validate.notNull(libraryIdentifier, "Library identifier is null.");
```